src/main/java/net/dv8tion/jda/internal/requests/restaction/GuildScheduledEventActionImpl.java

```diff
@@ -0,0 +1,178 @@
+    {
+        return (GuildScheduledEventActionImpl) super.timeout(timeout, unit);
+    }
+
+    @Nonnull
+    @Override
+    public GuildScheduledEventActionImpl deadline(long timestamp)
+    {
+        return (GuildScheduledEventActionImpl) super.deadline(timestamp);
+    }
+
+    @Nonnull
+    @Override
+    public Guild getGuild()
+    {
+        return guild;
+    }
+
+
+    @Nonnull
+    @Override
+    @CheckReturnValue
+    public GuildScheduledEventActionImpl setDescription(@Nullable String description)
+    {
+        if (description != null)
+            Checks.notLonger(description, GuildScheduledEvent.MAX_DESCRIPTION_LENGTH, "Description");
+        this.description = description;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public GuildScheduledEventAction setImage(@Nullable Icon icon)
+    {
+        this.image = icon;
+        return this;
+    }
+
+    @Override
+    protected RequestBody finalizeData()
+    {
+        preChecks();
+        DataObject object = DataObject.empty();
+        object.put("entity_type", entityType);
+        object.put("privacy_level", 2);
+        object.put("name", name);
+        object.put("scheduled_start_time", startTime.format(DateTimeFormatter.ISO_DATE_TIME));
+        if (entityType == 1 || entityType == 2)
+            object.put("channel_id", channelId);
+        if (entityType == 3 && location != null && location.length() > 0)
+            object.put("entity_metadata", DataObject.empty().put("location", location));
+        if (description != null)
+            object.put("description", description);
+        if (image != null)
+            object.put("image", image.getEncoding());
+        if (endTime != null)
+            object.put("scheduled_end_time", endTime.format(DateTimeFormatter.ISO_DATE_TIME));
+
+        return getRequestBody(object);
+    }
+
+    private void preChecks()
+    {
+        Checks.notNull(name, "Name");
+        Checks.notBlank(name, "Name");
+        Checks.notEmpty(name, "Name");
+        Checks.notLonger(name, GuildScheduledEvent.MAX_NAME_LENGTH, "Name");
+        Checks.notNull(startTime, "Start Time");
+        Checks.check(startTime.isAfter(OffsetDateTime.now()), "Cannot schedule event in the past!");
+
+        if (entityType == 3) {
+            Checks.notNull(location, "Location");
+            Checks.notBlank(location, "Location");
+            Checks.notEmpty(location, "Location");
+            Checks.notLonger(location, GuildScheduledEvent.MAX_LOCATION_LENGTH, "Location");
+            Checks.notNull(endTime, "End Time");
+            Checks.check((endTime).isAfter(startTime), "Cannot schedule event to end before starting!");
+
+        }
+
+    }
+
+
+    @Override
+    protected void handleSuccess(Response response, Request<GuildScheduledEvent> request)
+    {
+        request.onSuccess(api.getEntityBuilder().createGuildScheduledEvent((GuildImpl) guild, response.getObject()));
```

Comment:
This also adds it to cache

Reply:
should also be resolved now because the create method has a cacheflag check